### PR TITLE
Release version 2.17.0: "25 Din Mein Paisa Double"

### PR DESCRIPTION
**Pull request by jonpsy**
This automatically-generated pull request adds the commits necessary to make the 2.17.0 release.

Once the PR is merged, mlpack-bot will tag the release as HEAD~1 (so that it doesn't include the new HISTORY block) and publish it.

Or, well, hopefully that will happen someday.

When you merge this PR, be sure to merge it using a *rebase*.

### Changelog

 * CheckArbitraryFunctionTypeAPI extended for MOO support ([#283](https://github.com/mlpack/ensmallen/pull/283)).

 * Refactor NSGA2 ([#263](https://github.com/mlpack/ensmallen/pull/263), [#304](https://github.com/mlpack/ensmallen/pull/304)).

 * Add Indicators for Multiobjective optimizers ([#285](https://github.com/mlpack/ensmallen/pull/285)).

 * Make Callback flexible for MultiObjective Optimizers ([#289](https://github.com/mlpack/ensmallen/pull/289)).

 * Add ZDT Test Suite ([#273](https://github.com/mlpack/ensmallen/pull/273)).

 * Add MOEA-D/DE Optimizer ([#269](https://github.com/mlpack/ensmallen/pull/269)).

 * Introduce Policy Methods for MOEA/D-DE ([#293](https://github.com/mlpack/ensmallen/pull/293)).

 * Add Das-Dennis weight initialization method ([#295](https://github.com/mlpack/ensmallen/pull/295)).

 * Add Dirichlet Weight Initialization ([#296](https://github.com/mlpack/ensmallen/pull/296)).

 * Improved installation and compilation instructions ([#300](https://github.com/mlpack/ensmallen/pull/300)).

 * Disable building the tests by default for faster installation ([#303](https://github.com/mlpack/ensmallen/pull/303)).

 * Modify matrix initialisation to take into account default element zeroing in Armadillo 10.5 ([#305](https://github.com/mlpack/ensmallen/pull/305)).

### CODE REVIEW

**Comment by jonpsy** [comment]
"25 Din Mein Paisa Double" translates to "Money doubled in 25 days". A famous dialogue from the movie "Fir Hera Pheri". If you aren't already familiar with this meme, see [this video posted by Netfilx.](https://www.youtube.com/watch?v=0ew88RPjlK0&t=3s)

image

**Review by rcurtin — APPROVED** [review]
Awesome, looks like the script worked. :+1:

**Review by zoq — APPROVED** [review]
Looks good to me.

File: HISTORY.md [hunk 1]
```diff
@@ -1,14 +1,8 @@
...
-   default element zeroing in Armadillo 10.5
-   ([#305](https://github.com/mlpack/ensmallen/pull/305)).
 
- * Disable building the tests by default for faster installation
-   ([#303](https://github.com/mlpack/ensmallen/pull/303)).
-
- * Improved installation and compilation instructions
-   ([#300](https://github.com/mlpack/ensmallen/pull/300)).
+### ensmallen 2.17.0: "25 Din Mein Paisa Double"
+###### 2021-05-07
```
Comment (rcurtin) [inline HISTORY.md:5]: Actually, @jonpsy, I just realized, this date is backwards with the format we have been using in other places---this says May 7, but it should be `2021-07-05` (e.g. July 5).  Did you change anything in the script having to do with the date?

**Review by rcurtin — CHANGES_REQUESTED** [review]
Oops, just noticed an issue.  Because mlpack-bot does the release based on `HEAD~1`, probably the best option is to rerun the script for a new PR.

**Comment by conradsnicta** [comment]
@jonpsy Minor issue, though having a number (25) in the release name is potentially confusing with the version number (2.17).

Based on first hand experience with Armadillo, users can be trusted to make incorrect conclusions and do the wrong thing, no matter what the documentation says.

Perhaps there is a shorter version of "25 Din Mein Paisa Double" with similar intent, or maybe another quote from the "Fir Hera Pheri" movie?

**Comment by conradsnicta** [comment]
Minor issue, though having a number (25) in the release name is potentially confusing with the version number (2.17).

(Based on first hand experience with Armadillo, users can be trusted to make incorrect conclusions and do the wrong thing, treating the documentation as merely suggestions rather than as the canonical reference.)

Perhaps there is a shorter version of "25 Din Mein Paisa Double" with similar intent, or maybe another quote from the "Fir Hera Pheri" movie?

**Comment by zoq** [comment]
Replacing 25 with Pacchis, would make the title a little bit longer, from what @conradsnicta wrote above that might be too long?

**Comment by rcurtin** [comment]
Personally I don't think the length is a problem.  Hopefully nobody is actually *using* the version string for anything! :smile:

**Comment by jonpsy** [comment]
I can shorten other words, "Pacchis Din Me Pesa Double"  I think it's letter count should be more or less the same as the previous release (including space).

**Comment by coatless** [comment]
> Replacing 25 with Pacchis, would make the title a little bit longer, from what @conradsnicta wrote above that might be too long?

That should be fine. The previous name was 35 characters, c.f.

```r
R> nchar("Pacchis Din Mein Paisa Double")
[1] 29
R> nchar("Severely Dented Can Of Polyurethane")
[1] 35
```

**Comment by coatless** [comment]
> Personally I don't think the length is a problem. Hopefully nobody is actually _using_ the version string for anything! 😄

Not using the version string per say, but I do use the defines :)

File: HISTORY.md (hunk 1 above)
Comment (jonpsy) [reply HISTORY.md:5]: Ah my bad, didn't change the script. I just wrote in YYYY/DD/MM format by mistake in HISTORY.md

**Comment by conradsnicta** [comment]
> I can shorten other words, "Pachis Din Me Pesa Double" I think it's letter count should be more or less the same as the previous release (including space).

Sounds good :)

**Comment by rohitg00** [comment]
I think, You should try ITN staging for numbers to words.

**Pull request closed without merging**